File: c/1010.c

// https://www.urionlinejudge.com.br/judge/pt/problems/view/1010

#include <stdio.h>
#include <stdlib.h>

int main()
{
    int cod, num, i;
    double valor, pagar;
    pagar = 0.0f;

    for (i = 0; i < 2; i++){
        scanf("%d%d%lf", &cod, &num, &valor);

        pagar = pagar + (num * valor);

    }
    printf("VALOR A PAGAR: R$ %.2lf\n", pagar);
    return 0;
}

File: haskell/1010.hs

-- https://www.urionlinejudge.com.br/judge/pt/problems/view/1010

import Text.Printf (printf)

calcInputs :: IO Double
calcInputs = do input <- getLine
                let numbers = map read (words input) :: [Double]
                    cod = numbers !! 0
                    num = numbers !! 1
                    val = numbers !! 2
                return (num*val)

main :: IO ()
main = do r1 <- calcInputs
          r2 <- calcInputs
          printf "VALOR A PAGAR: R$ %.2f\n" (r1 + r2)